SampleApp/Program.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Text.Json;
+
+namespace SampleApp
+{
+    public class Program
+    {
+        /// <summary>
+        /// Session key for the cached forecast data
+        /// </summary>
+        private const string ForecastCacheKey = "forecast";
+
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            // Add services to the container.
+            builder.Services.AddAuthorization();
+
+            // Add session state support
+            builder.Services.AddSession(options =>
+            {
+                options.IdleTimeout = TimeSpan.FromSeconds(30);
+                options.Cookie.IsEssential = true;
+            });
+
+            // Add DynamoDB distributed cache for sessions
+            builder.Services.AddAWSDynamoDBDistributedCache(options =>
+            {
+                options.TableName = "weather-sessions-cache";
+                options.CreateTableIfNotExists = true;
```

Comment:
nit: Could you add a comment above this line explaining how we are auto creating the table for development purposes.

Reply:
Added a few lines with our disclaimer about production.